File: lintcode/reverse_integer.py
class Solution:
    # @param {int} n the integer to be reversed
    # @return {int} the reversed integer
    def reverseInteger(self, n):
        sign = 1 if n >= 0 else -1
        x = abs(n)
        result = 0
        while x > 0:
            result = result * 10 + x % 10
            x /= 10
        result *= sign
        return result if -2147483648 <= result <= 2147483647 else 0

File: leetcode/reverse_integer.js
/* Reverse digits of an integer.
 * Time Complexity: O(n)
 * Space Complexity: O(1)
 */
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    var sign = x >= 0 ? 1 : -1   
    x = Math.abs(x)
    for (var result = 0; x > 0; x = Math.floor(x / 10))
        result = result * 10 + x % 10
    result *= sign
    return -2147483648 <= result && result <= 2147483647 ? result : 0
};
